pkg/helper/secretutils.go

```diff
@@ -197,3 +204,147 @@ func IsSecretWatchedBy3scale(secret *v1.Secret) bool {
+	_, err = x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err == nil {
+		return nil
+	}
+	// If RSA parsings fails, try parsing as EC private key (PKCS#8 format)
+	_, err = x509.ParseECPrivateKey(block.Bytes)
+	if err == nil {
+		return nil
+	}
+	// Return an error if all parsing attempts fail.
+	return err
+}
+
+func ValidateRedisURL(url string) error {
+	redisPrefix := "rediss://"
+	// Check if the URL scheme is "rediss"
+	if !strings.HasPrefix(url, redisPrefix) {
+		return fmt.Errorf("URL should start with "+redisPrefix+". Found %s", url)
+	}
+	// Remove the "rediss://" prefix
+	url = strings.TrimPrefix(url, redisPrefix)
+	// Split the URL into host:port and optional database number
+	hostPortDB := strings.Split(url, "/")
+	if len(hostPortDB) > 2 {
+		return fmt.Errorf("URL has an invalid structure")
+	}
+	// Check if there's a database number part (e.g., after the slash)
+	dbPart := ""
+	if len(hostPortDB) == 2 {
+		dbPart = hostPortDB[1]
+	}
+	// Split the host and port part
+	hostPort := hostPortDB[0]
+	hostPortParts := strings.Split(hostPort, ":")
+	if len(hostPortParts) != 2 {
+		return fmt.Errorf("URL must contain a host and port, e.g., 'host:port'")
+	}
+	// Validate the host part (IP or Domain)
+	host := hostPortParts[0]
+	if net.ParseIP(host) == nil {
+		regex := `^(?i)([a-z0-9](-?[a-z0-9])*\.)+[a-z0-9](-?[a-z0-9])*$`
+		re := regexp.MustCompile(regex)
+		if !re.MatchString(host) {
+			return fmt.Errorf("Invalid host (neither IP address nor valid domain): %s", host)
+		}
+	}
+	// Validate the port (port part)
+	portStr := hostPortParts[1]
+	port, err := strconv.Atoi(portStr)
+	if err != nil || port < 1024 || port > 65535 {
+		return fmt.Errorf("Invalid port number: %s. Valid range is 1024-65535", portStr)
+	}
+	// Optionally, validate the database number (if it exists)
+	if dbPart != "" {
+		dbNum, err := strconv.Atoi(dbPart)
+		if err != nil || dbNum < 0 {
+			return fmt.Errorf("Invalid database number: %s", dbPart)
+		}
+	}
+	return nil
+}
+
+func ValidateRedisSentinelHostsForTLS(sentinelHosts string) error {
+	// 1. Validate Sentinel Hosts defined in system-redis or backend-redis secrets
+	// - if it's a valid to work with Redis Master in TLS mode:
+	// 1.1. Empty - Sentine is not defined. Redis Clients will work TLS Redis Master directly
+	// 1.2. One of Sentinel Hosts is TLS (rediss://...), - All can use TLS: Redis master, sentinel, clients
+	// 2. Not valid configurations:
+	// 2.1. One of sentinel hosts has wrong URL format
+	// 2.2. All Sentinel hosts are Non-TLS (redis://...). It's not valid as Master is TLS
+
+	if len(sentinelHosts) == 0 {
+		return nil
+	}
+	found := false
+	hostList := strings.Split(sentinelHosts, ",")
+	// Check if any of the Sentinel hosts uses 'rediss://' (TLS)
+	for _, host := range hostList {
+		host = strings.TrimSpace(host)
+		parsedURL, err := url.Parse(host)
+		if err != nil {
+			return fmt.Errorf("URL has an invalid structure: %s", host)
+		}
+		if parsedURL.Scheme == "rediss" {
```

Comment:
do we know for sure that if using sentinels with tls, every host will have rediss? 
are there any other scenarios where this can be set to a different value?
In general, I think I would just check if:
1. TLS required KEYS are present in secret
2. TLS required KEYS have non-empty VALUES

The rest of validation will be performed via preflight + components directly.

Reply:
regarding "every host will have rediss" - Yes, as for example. We want use TLS for client communication with Redis Server.  We have 3 Sentinel hosts for Backend (for HA), first - using TLS (rediss), - other - not.   If Sentinel is available - Client will not communicate with Redis Server directrly, it will use Sentinel. Secure Sentinel found, and communication established. Secure sentinel failed. Sentinel will do switch to "backup" Sentinel, but it's not secure. Communication - failed.

Reply:
validation moved to HA, and only basic validation done now. Thank you